From d1bde7a137c9dac74018327944ae96029fe158f0 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 28 Sep 2022 17:51:48 +0300
Subject: [PATCH] [SW-105911] common: move start addr to global data structure

Currently, startaddr of ppboot which is required to jump to
ppboot in case of validation pass flag is set ie. ROM doesn't perform
dma and sha verification of image is store in non-persistent ROM
structure. This patch moves startaddr to persistent ROM structure
which is retained across non-cold reset.

Change-Id: I0bcbeb0df16a41e81a865903aefc5eeea8d0747c
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index b8e2278..73a6185 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -932,8 +932,7 @@
 void arcboot_rom_bringup_ppboot(void)
 {
 	void (*ppboot)();
-	uint64_t *start_addr =
-			&arcboot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_SPI].start_addr;
+	uint64_t *start_addr = &arcboot_rom_global.ppboot_start_addr;
 	int ret;
 
 	if (!arcboot_rom_boot_desc_init_done(HL_FW_BOOT_PTR_SPI)) {
diff --git a/common/hl_boot_desc_rom.h b/common/hl_boot_desc_rom.h
index 81af8fa..e7b63b9 100644
--- a/common/hl_boot_desc_rom.h
+++ b/common/hl_boot_desc_rom.h
@@ -28,7 +28,6 @@
 struct hl_fw_boot_ptrs_ctx {
 	uint8_t boot_desc_flag;
 	uint32_t img_ptr_spi_offset;
-	uint64_t start_addr;
 	struct hl_fw_desc_info desc_info;
 	uint64_t dest_addr_dma;
 	uint32_t dest_addr_arc;
diff --git a/include/arcboot.h b/include/arcboot.h
index a421d99..e6fb0e6 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -104,6 +104,7 @@
 	uint32_t d0_fuse_spi_offset;
 	uint32_t d1_fuse_spi_offset;
 	bool fuse_spi;
+	uint64_t ppboot_start_addr;
 	struct hl_fw_sec_info sec_data;
 	struct hl_fw_boot_ptrs spi_manifest_hdr;
 };
